addons/website/controllers/main.py

```diff
@@ -333,6 +335,193 @@ def get_dynamic_snippet_templates(self, filter_name=False):
+        """
+        order = self._get_search_order(order)
+        options = options or {}
+        results_count, search_results, fuzzy_term = request.website._search_with_fuzzy(search_type, term, limit, order, options)
+        if results_count == 0:
+            return {
+                'results': [],
+                'results_count': 0,
+                'parts': {},
+            }
+        term = fuzzy_term or term
+        search_results = request.website._search_render_results(search_results, limit)
+
+        mappings = []
+        results_data = []
+        for search_result in search_results:
+            results_data += search_result['results_data']
+            mappings.append(search_result['mapping'])
+        if search_type == 'all':
+            # Only supported order for 'all' is on name
+            results_data.sort(key=lambda r: r.get('name', ''), reverse='name desc' in order)
+        results_data = results_data[:limit]
+        result = []
+        for record in results_data:
+            mapping = record['_mapping']
+            mapped = {
+                '_fa': record.get('_fa'),
+            }
+            for mapped_name, field_meta in mapping.items():
+                value = record.get(field_meta.get('name'))
+                if value:
+                    field_type = field_meta.get('type')
+                    if field_type == 'text':
+                        if value and len(value) > max_nb_chars:
+                            value = "%s..." % value[:(max_nb_chars - 3)]
+                        if field_meta.get('match') and value and term:
+                            pattern = '|'.join([re.escape(search_term) for search_term in term.split()])
+                            if pattern:
+                                parts = re.split('(%s)' % pattern, value, flags=re.I)
+                                if len(parts) > 1:
+                                    value = request.env['ir.ui.view'].sudo()._render_template(
+                                        "website.search_text_with_highlight",
+                                        {'parts': parts}
+                                    ).decode('utf-8')
+                                    field_type = 'html'
+
+                    if (field_type in ('image', 'binary')):
+                        value = escape(value)
+                    elif ('ir.qweb.field.%s' % field_type) in request.env:
+                        opt = {}
+                        if field_type == 'monetary':
+                            opt['display_currency'] = options['display_currency']
+                        elif field_type == 'html':
+                            opt['template_options'] = {}
+                        value = request.env[('ir.qweb.field.%s' % field_type)].value_to_html(value, opt)
+                    else:
+                        value = escape(value)
+                else:
+                    value = ''
+                mapped[mapped_name] = Markup(value)
```

Comment:
Doesn't seem useful? It's not used on literal markup contents, and it's going to be stripped by the serialization.

Reply:
`value` is an "already escaped/rendered" value, it can be the result of a field rendering and monetary contains `<span>`.
The result is used in two ways:
- from the client through RPC, in which case the client re-markups the fields
- from the server when it renders the general search results page: in that case it sends those results into a QWeb rendering on the server side (route "/website/search") - in that case the Markup is needed to avoid double-escaping

Reply:
> from the server when it renders the general search results page: in that case it sends those results into a QWeb rendering on the server side (route "/website/search") - in that case the Markup is needed to avoid double-escaping

That should not be an issue though:

* qweb should always mark its output as markup-safe
* the escape function does it (it's an alias to `markupsafe.Markup`)
* `value_to_html` should also be marking its output as safe internally

The only situation I can see where this would be an issue is a `value_to_html` which returns HTML but doesn't properly mark it.

Reply:
This is indeed the case for the `html` field type, but maybe there is a good reason not to make it markup-safe:
```py
class HTMLConverter(models.AbstractModel):
...
    def value_to_html(self, value, options):
...
        return etree.tostring(body, encoding='unicode', method='html')[6:-7]
```

Reply:
@bso-odoo it looks like that was originally missed / missing, and @Gorash fixed it in the qweb rewriting branch: the current master does flag this value as `Markup`. So a rebase would fix that issue.

Reply:
Great ! It works ! No more Markup on server side.
Only the one on the client side remains - and it now triggers an alert.

Reply:
Yeah but the client side seems fine / correct so it'll be easy to override.